lib/websocket_server.rb

```diff
@@ -15,7 +15,10 @@ def initialize
         end
 
         # Skip this loop if we can't do anything
-        next if Array(reads).empty? || writes.empty? || errors.any?
+        if Array(reads).empty? || writes.empty? || errors.any?
```

Comment:
`It will return nil if optional timeout value is given and no IO object is ready in timeout seconds.`
from http://ruby-doc.org/core-2.2.0/IO.html